cgo_source.go

```diff
@@ -1,4 +1,4 @@
-//go:build !duckdb_use_lib && (duckdb_from_source || !(darwin || (linux && amd64)))
```

Comment:
I think we should add `arm64` alongside `amd64` as there are plenty of other platforms on linux, we currently do not support (Mips, RISC-V)...

Reply:
I changed `(linux && amd64)` to `(linux && (amd64 || arm64))` to limit it to only those two architectures.